src/Algebra/Properties/CancellativeCommutativeSemiring.agda

```diff
@@ -0,0 +1,72 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some theory for CancellativeCommutativeSemiring.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra using (CancellativeCommutativeSemiring)
+import Algebra.GCD
+import Algebra.Primality
+import Algebra.Properties.Semigroup.Divisibility as SemigroupDiv
+import Algebra.Properties.Semiring.Divisibility as SemiringDiv
+open import Data.Product using (_×_; _,_; proj₁; proj₂)
+open import Data.Sum using (_⊎_; inj₁; inj₂)
+open import Function using (case_of_; flip)
+open import Level using (_⊔_)
+open import Relation.Binary using (Rel; Symmetric; Setoid)
+open import Relation.Unary using (Pred)
+
+module Algebra.Properties.CancellativeCommutativeSemiring
+  {α ℓ} (R : CancellativeCommutativeSemiring α ℓ)
+  where
+
+open CancellativeCommutativeSemiring R renaming (Carrier to A)
+
+open SemigroupDiv *-semigroup using (_∣_; _∤_; ∣-trans)
+--
+-- Reexporting Divisibility over Semigroup, Semiring
+--
+open SemiringDiv semiring using (∣nonzero⇒≉0)
```

Comment:
This should be opened `public`ly and everything should be re-exported. Not just `∣nonzero⇒≉0`. It should also go underneath the opening of `Algebra.Primality` and `Algebra.GCD`.

Reply:
I set this as 
```
open Algebra.Primality _≈_ _*_ 0# 1# using (Irreducible; irredᶜ; Coprime)
open Algebra.GCD _≈_ _*_ using (GCD)
open SemigroupDiv *-semigroup public
--
-- Reexporting divisibility over Semigroup, Semiring
--
open SemiringDiv semiring public using (∣nonzero⇒≉0)
```
(as I understand you, this commentary needs to change to your style).

If I delete the last `using`, then there is an overlap with
``open SemigroupDiv *-semigroup public``.

Please, advise.